packages/loopring_v3/contracts/impl/libexchange/ExchangeConstants.sol

```diff
@@ -26,49 +27,37 @@ library ExchangeConstants
     /// @dev Returns a list of constants used by the exchange.
     /// @return constants The list of constants in the following order:
     ///         SNARK_SCALAR_FIELD
-    ///         MAX_PROOF_GENERATION_TIME_IN_SECONDS
-    ///         MAX_GAP_BETWEEN_FINALIZED_AND_VERIFIED_BLOCKS
     ///         MAX_OPEN_DEPOSIT_REQUESTS
     ///         MAX_OPEN_WITHDRAWAL_REQUESTS
-    ///         MAX_AGE_UNFINALIZED_BLOCK_UNTIL_WITHDRAW_MODE
     ///         MAX_AGE_REQUEST_UNTIL_FORCED
     ///         MAX_AGE_REQUEST_UNTIL_WITHDRAW_MODE
     ///         MAX_TIME_IN_SHUTDOWN_BASE
     ///         MAX_TIME_IN_SHUTDOWN_DELTA
     ///         TIMESTAMP_HALF_WINDOW_SIZE_IN_SECONDS
     ///         MAX_NUM_TOKENS
     ///         MAX_NUM_ACCOUNTS
-    ///         MAX_TIME_TO_DISTRIBUTE_WITHDRAWALS
-    ///         MAX_TIME_TO_DISTRIBUTE_WITHDRAWALS_SHUTDOWN_MODE
     ///         FEE_BLOCK_FINE_START_TIME
     ///         FEE_BLOCK_FINE_MAX_DURATION
-    ///         MIN_GAS_TO_DISTRIBUTE_WITHDRAWALS
     ///         MIN_AGE_PROTOCOL_FEES_UNTIL_UPDATED
     ///         GAS_LIMIT_SEND_TOKENS
     function getConstants()
         external
         pure
-        returns(uint[20] memory)
+        returns(uint[14] memory)
```

Comment:
should we return a struct? Is it possible with the new compiler?

Reply:
Done. I thought it may give problems when more than 16 values are in the struct (because of stack limitations that for some reason solidity still doesn't properly implement), but even if that's the case you can actually work around that easily by assigning the struct values one by one instead of all at once using the stack.